### ISSUE TITLE
gulp.compass injects css dir at root

### ISSUE THREAD

**Issue by jkauszler** [planetoftheweb/workflows#1]
Hello and thanks for the tutorial! 

I'm following along and noticed that whenever the gulp.task 'compass' is performed it creates an addition 'css' directory at the root.

Any idea why this is happening and how to prevent it?

Thanks!

**Comment by planetoftheweb** [comment]
I think it has to do with the setup for gulp. In the same way you specify a directory for your HTML and IMAGES, if you add one for your CSS, you can control where it appears in your directory structure.—

**Comment by rphuber** [comment]
It seems that this is already the case.  I too have experienced this problem and get it in the specified dest as well as in the root file.  Also I see that it also happens in your video.  I checked over the gulp file and have this line:
.pipe(gulp.dest('builds/development/css'))
any other recommendations?  Thanks!

**Comment by jkauszler** [comment]
Hey, guys.  Try switching over from using the gulp-compass plugin (over whatever plugin is used in the tutorial) to gulp-ruby-sass (https://www.npmjs.org/package/gulp-ruby-sass/).  If I remember correctly setup is nearly identical and it doesn't inject an extra dir.

**Comment by planetoftheweb** [comment]
That's true. I think gulp-sass also doesn't require ruby. I'm not sure if
it processes compass though.

**Comment by JCoe77** [comment]
I think I was able to find a solution to this issue without switching to a different plugin. The gulp-compass plugin accepts a css option; https://www.npmjs.com/package/gulp-compass#css

So with that said, I modified my compass task to look like this:

``` javascript
gulp.task('compass', function() {
  gulp.src(sassSources)
    .pipe(compass({
      css: outputDir + 'css',
      sass: 'components/sass',
      image: outputDir + 'images',
      style: sassStyle
    })
    .on('error', gutil.log))
    //.pipe(gulp.dest(outputDir + 'css'))
    .pipe(connect.reload())
});
```

Now the compass plugin is writing the generated output to the proper output directory and there is no duplicate directory created.

**Comment by AjGabriel** [comment]
Hello,

I am a complete newbie and am trying to emulate the course using my Mac OS X Yosemite.

With the current dependencies that I have, it is far more updated that one's listed here. All have been running well up until I came along with the task for compass. I keep running on errors like:

_Error installing compass:
    ERROR: Failed to build gem native extension._

but then again I installed the xcode CLI and the compass and ruby installed well. However moving along the course, when I am trying to execute the gulp.task compass, I got this error:

_module.js:327
    throw err;
    ^

Error: Cannot find module 'gulp-compass'_

Here are the codes that I did:

```
_sudo npm install --save-dev gulp compass_
```

I get warnings that it is deprecated I was unable to move along with the course.

I know this is a completely newbie question. Please help.
Thanks.

**Comment by jkauszler** [comment]
AjGabriel - I'm not sure that I have enough information to pin point the problem, but it definitely appears that compass is not installed.  I tried googling:

`Error installing compass:
ERROR: Failed to build gem native extension.`

and this is what I found that looked relevant. http://stackoverflow.com/questions/22544754/failed-to-build-gem-native-extension-installing-compass If this doesn't work, I would try doing the same and see what you find. It's almost assuredly on Stack Overflow somewhere.
